tests/foreman/virtwho/api/test_esx_sca.py

```diff
@@ -0,0 +1,500 @@
+
+
+@pytest.fixture()
+def form_data(module_sca_manifest_org, target_sat):
+    form = {
+        'name': gen_string('alpha'),
+        'debug': 1,
+        'interval': '60',
+        'hypervisor_id': 'hostname',
+        'hypervisor_type': settings.virtwho.esx.hypervisor_type,
+        'hypervisor_server': settings.virtwho.esx.hypervisor_server,
+        'organization_id': module_sca_manifest_org.id,
+        'filtering_mode': 'none',
+        'satellite_url': target_sat.hostname,
+        'hypervisor_username': settings.virtwho.esx.hypervisor_username,
+        'hypervisor_password': settings.virtwho.esx.hypervisor_password,
+    }
+    return form
+
+
+@pytest.fixture()
+def virtwho_config(form_data, target_sat):
+    virtwho_config = target_sat.api.VirtWhoConfig(**form_data).create()
+    yield virtwho_config
+    virtwho_config.delete()
+    assert not target_sat.api.VirtWhoConfig().search(query={'search': f"name={form_data['name']}"})
+
+
+class TestVirtWhoConfigforEsx:
+    @pytest.mark.tier2
+    @pytest.mark.parametrize('deploy_type', ['id', 'script'])
+    def test_positive_deploy_configure_by_id_script(
+        self, module_sca_manifest_org, form_data, virtwho_config, target_sat, deploy_type
+    ):
+        """Verify "POST /foreman_virt_who_configure/api/v2/configs"
+
+        :id: 26b37372-390f-45b8-accd-ff006a8b0ccf
+
+        :expectedresults: Config can be created and deployed
+
+        :CaseLevel: Integration
+
+        :CaseImportance: High
+        """
+        assert virtwho_config.status == 'unknown'
+        if deploy_type == "id":
+            command = get_configure_command(virtwho_config.id, module_sca_manifest_org.name)
+            deploy_configure_by_command(
+                command, form_data['hypervisor_type'], debug=True, org=module_sca_manifest_org.label
+            )
+        elif deploy_type == "script":
+            script = virtwho_config.deploy_script()
+            deploy_configure_by_script(
+                script['virt_who_config_script'],
+                form_data['hypervisor_type'],
+                debug=True,
+                org=module_sca_manifest_org.label,
+            )
+        virt_who_instance = (
+            target_sat.api.VirtWhoConfig()
+            .search(query={'search': f'name={virtwho_config.name}'})[0]
+            .status
+        )
+        assert virt_who_instance == 'ok'
+
+    @pytest.mark.tier2
+    def test_positive_debug_option(
+        self, module_sca_manifest_org, form_data, virtwho_config, target_sat
+    ):
+        """Verify debug option by "PUT
+
+        /foreman_virt_who_configure/api/v2/configs/:id"
+
+        :id: 61eeef42-12a0-4b92-9b87-c409b2507052
+
+        :expectedresults: debug option can be updated.
+
+        :CaseLevel: Integration
+
+        :CaseImportance: Medium
+        """
+        options = {'true': '1', 'false': '0', '1': '1', '0': '0'}
+        for key, value in sorted(options.items(), key=lambda item: item[0]):
```

Comment:
instead of sorting a static dictionary at runtime, try instead to define the dictionary in the order you want. Then you just need to iterate over an already sorted dictionary.

Reply:
> instead of sorting a static dictionary at runtime, try instead to define the dictionary in the order you want. Then you just need to iterate over an already sorted dictionary.

@JacobCallahan Have realdy updated the code with order 
```
options = {'0': '0', '1': '1', 'false': '0', 'true': '1'}
        for key, value in options.items():
```